grr/lib/data_stores/http_data_store.py

```diff
@@ -83,7 +83,10 @@ def _ReadExactly(self, n):
     ret = ""
     left = n
     while left:
-      ret += self.sock.recv(left)
+      data = self.sock.recv(left)
+      if data == "":
+        raise IOError("Expected %d bytes, got EOF after %d", n, len(ret))
```

Comment:
maybe it makes more sense to raise socket.error so it will return none in readreply. @grrrrrrrrr you know this code better than me, can you take a look?